### PR TITLE
feat(webpack-cli): allow multiple entry files

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
YES
**If relevant, did you update the documentation?**
YES
**Summary**
Now we can provide multiple entry files via CLI using  -
- `webpack --entry=a.js --entry=b.js` 
- `webpack ./a.js ./b.js`

Refers #717
Fixes #1454 

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

**Comment by snitin315** [comment]
> webpack src/one.js src/two.js src/three.js

@evilebottnawi we can do this for entry but what about other flags that can take multiple values?

**Comment by evilebottnawi** [comment]
@snitin315 Let's start with entry, it is most popular use case

File: packages/webpack-cli/lib/bootstrap.js [hunk 1]
```diff
@@ -49,14 +49,21 @@ async function runCLI(cli, commandIsUsed) {
...
-                parsedArgs.unknownArgs
-                    .filter((e) => e)
-                    .forEach((unknown) => {
-                        logger.warn('Unknown argument:', unknown);
+            if (parsedArgs.unknownArgs.length > 0 && !parsedArgs.unknownArgs[0].startsWith('-')) {
+                if (parsedArgs.unknownArgs.length === 1) {
+                    entry = parsedArgs.unknownArgs[0];
+                } else {
+                    entry = [];
+                    parsedArgs.unknownArgs.forEach((unknown) => {
```
Comment (rishabh3112) [inline packages/webpack-cli/lib/bootstrap.js:57]: `webpack --entry a.js b.js c.js`  is what we should be supporting imo.

Comment (snitin315) [reply packages/webpack-cli/lib/bootstrap.js:57]: It will be a bit tricky to parse `--entry` like  that.

Comment (snitin315) [reply packages/webpack-cli/lib/bootstrap.js:57]: We need to discuss which to support or support both.
/cc @webpack/cli-team

File: packages/webpack-cli/lib/bootstrap.js [hunk 2]
```diff
@@ -49,14 +49,21 @@ async function runCLI(cli, commandIsUsed) {
...
             // an unknown flag rather than an entry
             let entry;
-            if (parsedArgs.unknownArgs.length === 1 && !parsedArgs.unknownArgs[0].startsWith('-')) {
-                entry = parsedArgs.unknownArgs[0];
-            } else if (parsedArgs.unknownArgs.length > 0) {
-                parsedArgs.unknownArgs
-                    .filter((e) => e)
-                    .forEach((unknown) => {
-                        logger.warn('Unknown argument:', unknown);
+            if (parsedArgs.unknownArgs.length > 0 && !parsedArgs.unknownArgs[0].startsWith('-')) {
```
Comment (evenstensberg) [inline packages/webpack-cli/lib/bootstrap.js:52]: This is not a way to go, set multipleValues in cliFlags instead

Comment (snitin315) [reply packages/webpack-cli/lib/bootstrap.js:52]: That way we will be able to support  `webpack --entry a.js --entry b.js`

The idea is to support multiple entries by default without passing `--entry` i.e. `webpack a.js b.js c.js`

Comment (snitin315) [reply packages/webpack-cli/lib/bootstrap.js:52]: Or we can support , need discussion for same.

File: packages/webpack-cli/lib/bootstrap.js (hunk 1 above)
Comment (evilebottnawi) [reply packages/webpack-cli/lib/bootstrap.js:57]: @rishabh3112 I think we support two cases here for better 0CJS:
```
webpack --entry ./src/index.js --entry ./src/other.js
webpack ./src/index.js ./src/other.js
```

Second just sugar.

Comment (snitin315) [reply packages/webpack-cli/lib/bootstrap.js:57]: I agree with @evilebottnawi we should support both.

Comment (rishabh3112) [reply packages/webpack-cli/lib/bootstrap.js:57]: Lets do both then. But this would be a breaking change so should be mentioned as such during release.

Comment (evilebottnawi) [reply packages/webpack-cli/lib/bootstrap.js:57]: @rishabh3112 Why? It is breaking change?

Comment (rishabh3112) [reply packages/webpack-cli/lib/bootstrap.js:57]: Yeah it's not, my bad. I wrongly thought that we were supporting configurations to be passed without flags instead on entry in CLI v3.

Comment (evilebottnawi) [reply packages/webpack-cli/lib/bootstrap.js:57]: Configuration should loaded by default (0CJS) or using `--config` flag

**Comment by snitin315** [comment]
Ready for review :raised_hands:

**Comment by octaharon** [comment]
Hey lads. What about named entries? Webpack 4 allowed passing `<name>=request>` as entry, as of now it doesn't seem to work, and "--entry" argument is treated like a filepath regardless of format. There seems to be no deprecation warning about it, will this functionality be restored?

**Comment by petemill** [comment]
> Hey lads. What about named entries? Webpack 4 allowed passing <name>=<request> as entry, as of now it doesn't seem to work, and "--entry" argument is treated like a filepath regardless of format. There seems to be no deprecation warning about it, will this functionality be restored?

Confirming this is broken now in webpack-cli. You cannot define an entry's name in the cli, preventing different compiled file names than entry file names, where this is possible in webpack config.

**Pull request merged**